types/ember-cli-fastboot/test/ember-cli-fastboot-tests.ts

```diff
@@ -0,0 +1,23 @@
+import { assertType } from './assert';
+import { FastBoot as FastBootService, FastbootRequest, Shoebox } from 'ember-cli-fastboot/services/fastboot';
+import { FastBoot as _FastBoot } from 'ember-cli-fastboot/-private';
+
+/** type assertions for global FastBoot object */
+assertType<_FastBoot | undefined>(FastBoot);
+
+/** type assertions for fastboot service & other classes */
+const instance = FastBootService.create();
+assertType<FastbootRequest>(instance.request);
+assertType<boolean>(instance.isFastboot);
+assertType<Shoebox>(instance.shoebox);
+instance.deferRendering(new Promise<'foo'>(() => 'foo')); // $ExpectType void
+
+instance.shoebox.put('foo', 'bar'); // $ExpectType void
+assertType<unknown>(instance.shoebox.retrieve('foo'));
```

Comment:
This is a tricky one: `assertType` is a *very* dumb kind of assertion, so it will actually accept *anything* here. The reason is that `unknown` is the “*top*” type in the type system: you can assign *anything* to it legally. (The opposite is `never`, which you *cannot* assign anything to, and is the “*bottom*” of the type system.)

Happily `// $ExpectType` will actually catch this because it doesn't just check for assignability, but whether the type actually matches.

```suggestion
instance.shoebox.retrieve('foo'); // $ExpectType unknown
```

---

As an aside: this is part of why we use [the `expect-type` library](https://github.com/mmkal/ts/tree/master/packages/expect-type) for types published from packages in the Ember ecosystem: it *does* correctly handle this:

```
// ✅
expectTypeOf(instance.shoebox.retrieve('foo')).toEqualTypeOf<unknown>();

// ❌
expectTypeOf("hello").toEqualTypeOf<unknown>();
```